### PR TITLE
Add: lambda(), make a chain of filters reusable

### PR DESCRIPTION
**Pull request by Kijewski**

### CODE REVIEW

**Comment by simonask** [comment]
Out of curiosity, is there precedence for the name `lambda` in other ranges libraries to achieve this?

An alternative name that might be more immediately clear could be `defer`.

Also, have you considered putting the `lambda()` at the end instead of the beginning of the expression? This might be more natural with the `defer` naming, but it seems this would avoid having to overload `operator|` specially.

**Comment by Kijewski** [comment]
> Out of curiosity, is there precedence for the name lambda in other ranges libraries to achieve this?

I don't know. I chose the name because lamda() makes the expression act like a lambda function.

> Also, have you considered putting the lambda() at the end instead of the beginning of the expression?

I would not like that. With the current implementation you can tell were the value will get injected: at the front. And it kinda reads a a lambda expression in lambda calculus … at least a little bit.

I don't know how one could achieve this functionality if it was put at the end. operator| is left-associative.

File: include/rx/ranges.hpp [hunk 1]
```diff
@@ -2317,6 +2325,60 @@ template <class... Inputs>
...
 
+/// Make chained operations reusable.
+template <class... Fs>
+struct lambda : detail::lambda_tag {
+    std::tuple<Fs...> funcs;
+
+    constexpr lambda() = default;
+
+    template <class... Fx>
+    constexpr lambda(Fx&&... funcs) : funcs(std::forward<Fx>(funcs)...) {}
```
Comment (Hazurl) [inline include/rx/ranges.hpp:2336]: This prevents the use of the copy/move constructor

**Comment by Hazurl** [comment]
I would also suggest changing the name...

**Comment by Kijewski** [comment]
Maybe `placeholder()`, like [std::placeholders](https://en.cppreference.com/w/cpp/utility/functional/placeholders)?

**Comment by simonask** [comment]
Ah, now I see the aim here.

The following will work just as well, though:

```
auto f = [](auto&& input) {
    return input | take(...) | filter(...);
};
auto result = seq() | f | to_vector();
```

Unless there is another compelling reason that I'm not seeing, I have to say I'm not in favor of introducing alternative ways to write lambda functions. :-)

**Comment by Kijewski** [comment]
I guess you're right. A "normal" lambda function is easier to read then the lambda() variant, and at least spontaneously I cannot find a compelling reason why this function is needed.

**Pull request closed without merging**